source/declarations.tex

```diff
@@ -2526,15 +2529,19 @@
 \item
 Otherwise, let $S$ be the entity associated with the scope inhabited by
 the \grammarterm{declarator}.
+\end{itemize}
+Then:
+\begin{itemize}
 \item
 If the \grammarterm{declarator} declares
 an explicit instantiation or a partial or explicit specialization,
 the \grammarterm{declarator} does not bind a name.
-If it declares a class member,
-the terminal name of the \grammarterm{declarator-id} is not looked up;
-otherwise, only those lookup results that are nominable in $S$
+If such a \grammarterm{declarator} does not declare a class member,
```

Comment:
@opensdh I tried to figure out why we're treating class members differently from non-class-members here and didn't succeed. Perhaps an example would help? I'm also not sure how we're supposed to handle an explicit specialization of a class member template without performing name lookup. Was the intent here instead that if the specialization or instantiation is for a non-template member of a class template specialization, no lookup is performed?

Reply:
I think that you're right: we do have to do the lookup for the same reason that we have to do it when encountering `template<> my_namespace::my_class<` since it might continue `int>::sub_template<int> {` or just `int> {`.  You're probably right about the intent too, but I'm less clear on that at the moment.  (My plan is to, tomorrow, look at _your_ wording here and see if I see any problems with it of a similar sort; if not, it stands to reason that it reflects the intent better.)